### DISCUSSION TITLE
Creating ViewModels with Mapperly

### DISCUSSION THREAD

**Discussion by BicycleMark** [riok/mapperly#1475] [General]
Are there mapperly users mapping from ef core classes to ViewModels that expose public properties as INotifyPropertyChange type properties? Is that a Bad practice?

So far my research indicates that:
1) we cannot use MvvmCommunity Toolkit [ObservableProperies] on a private field - due to fact that chaining generators is not supported
2) It seems that we can Map from a Public ef core Property to a Private field on the Dto object. So I am thinking we could expose in the Dto a public property that implements the getters / setters in a traditional INotifyProoperty manner.

**Comment by latonz** [comment]
1. As you know already chaining source generators is currently not supported by .NET. See #1445, #883, #292, #533, [FAQ](https://mapperly.riok.app/docs/getting-started/faq/#mapperly-does-not-work-when-i-use-source-generator-x) and https://github.com/dotnet/roslyn/issues/57239.
2. You could either embed the Mapper in the DTO so that the member can safely access the private members (may break some separation principles, depending on your code architecture), or you could use [private member mapping](https://mapperly.riok.app/docs/configuration/private-member-mapping/) (which comes with its own limitations...).
